contracts/modules/RelayerModule.sol

```diff
@@ -0,0 +1,372 @@
+    {
+        return keccak256(
+            abi.encodePacked(
+                "\x19Ethereum Signed Message:\n32",
+                keccak256(abi.encodePacked(
+                    byte(0x19),
+                    byte(0),
+                    _from,
+                    _to,
+                    _value,
+                    _data,
+                    _nonce,
+                    _gasPrice,
+                    _gasLimit,
+                    _refundToken,
+                    _refundAddress))
+        ));
+    }
+
+    /**
+    * @dev Checks if the relayed transaction is unique. If yes the state is updated.
+    * For actions requiring 1 signature by the owner we use the incremental nonce.
+    * For all other actions we check/store the signHash in a mapping.
+    * @param _wallet The target wallet.
+    * @param _nonce The nonce.
+    * @param _signHash The signed hash of the transaction.
+    * @param requiredSignatures The number of signatures required.
+    * @param ownerSignatureRequirement The wallet owner signature requirement.
+    * @return true if the transaction is unique.
+    */
+    function checkAndUpdateUniqueness(
+        address _wallet,
+        uint256 _nonce,
+        bytes32 _signHash,
+        uint256 requiredSignatures,
+        OwnerSignature ownerSignatureRequirement
+    )
+        internal
+        returns (bool)
+    {
+        if (requiredSignatures == 1 && ownerSignatureRequirement == OwnerSignature.Required) {
+            // use the incremental nonce
+            if (_nonce <= relayer[_wallet].nonce) {
+                return false;
+            }
+            uint256 nonceBlock = (_nonce & 0xffffffffffffffffffffffffffffffff00000000000000000000000000000000) >> 128;
+            if (nonceBlock > block.number + BLOCKBOUND) {
+                return false;
+            }
+            relayer[_wallet].nonce = _nonce;
+            return true;
+        } else {
+            // use the txHash map
+            if (relayer[_wallet].executedTx[_signHash] == true) {
+                return false;
+            }
+            relayer[_wallet].executedTx[_signHash] = true;
+            return true;
+        }
+    }
+
+    /**
+    * @dev Validates the signatures provided with a relayed transaction.
+    * The method MUST throw if one or more signatures are not valid.
+    * @param _wallet The target wallet.
+    * @param _signHash The signed hash representing the relayed transaction.
+    * @param _signatures The signatures as a concatenated byte array.
+    * @param _option An enum indicating whether the owner is required, optional or disallowed.
+    * @return A boolean indicating whether the signatures are valid.
+    */
+    function validateSignatures(
+        address _wallet,
+        bytes32 _signHash,
+        bytes memory _signatures,
+        OwnerSignature _option
+    )
+        internal
+        view
+        returns (bool)
+    {
+        if (_signatures.length == 0) {
+            return true;
```

Comment:
Seems to me that it's equivalent to what we had before where we would skip `validateSignature` when `requiredSignatures == 0` (We had `require(requiredSignatures == 0 || validateSignatures(_wallet, signHash, _signatures, ownerSignatureRequirement)`)

Reply:
We used to skip the signature validation when `requiredSignatures` was 0. This basically embeds that check in the `validateSignature` method to keep the code of `execute` cleaner. 

Note that this check is not necessary and can be removed since with `signatures.length = 0` the method always returns `true`. But it avoids a few gas consuming steps (in particular loading the guardians in L277).